packages/netlify-cms-core/src/components/MediaLibrary/MediaLibrary.js

```diff
@@ -210,6 +210,36 @@ class MediaLibrary extends React.Component {
     });
   };
 
+  /**
+   * Downloads the selected file.
+   */
+  handleDownload = () => {
+    const { selectedFile } = this.state;
+    console.log(selectedFile);
+    if (!selectedFile.url) {
+      return;
+    }
```

Comment:
I think this will fix the bugs I mentioned
```suggestion
    const { displayURLs } = this.props;
    const url = displayURLs.getIn([selectedFile.id, 'url']) || selectedFile.url;
    if (!url) {
      return;
    }
```